XamarinCommunityToolkit/Behaviors/TextValidationBehavior.shared.cs

```diff
@@ -0,0 +1,110 @@
+
+        public static readonly BindableProperty MaximumLengthProperty = BindableProperty.Create(nameof(MaximumLength), typeof(int), typeof(TextValidationBehavior), int.MaxValue, propertyChanged: OnValidationPropertyChanged);
+
+        public static readonly BindableProperty ShouldTrimValueProperty = BindableProperty.Create(nameof(ShouldTrimValue), typeof(bool), typeof(TextValidationBehavior), false, propertyChanged: OnValidationPropertyChanged);
+
+        public static readonly BindableProperty ShouldConvertNullToEmptyValueProperty = BindableProperty.Create(nameof(ShouldConvertNullToEmptyValue), typeof(bool), typeof(TextValidationBehavior), true, propertyChanged: OnValidationPropertyChanged);
+
+        public static readonly BindableProperty RegexPatternProperty = BindableProperty.Create(nameof(RegexPattern), typeof(string), typeof(TextValidationBehavior), defaultValueCreator: GetDefaultRegexPattern, propertyChanged: OnRegexPropertyChanged);
+
+        public static readonly BindableProperty RegexOptionsProperty = BindableProperty.Create(nameof(RegexOptions), typeof(RegexOptions), typeof(TextValidationBehavior), defaultValueCreator: GetDefaultRegexOptions, propertyChanged: OnRegexPropertyChanged);
+
+        Regex regex;
+
+        public int MinimumLength
+        {
+            get => (int)GetValue(MinimumLengthProperty);
+            set => SetValue(MinimumLengthProperty, value);
+        }
+
+        public int MaximumLength
+        {
+            get => (int)GetValue(MaximumLengthProperty);
+            set => SetValue(MaximumLengthProperty, value);
+        }
+
+        public bool ShouldTrimValue
+        {
+            get => (bool)GetValue(ShouldTrimValueProperty);
+            set => SetValue(ShouldTrimValueProperty, value);
+        }
+
+        public bool ShouldConvertNullToEmptyValue
+        {
+            get => (bool)GetValue(ShouldConvertNullToEmptyValueProperty);
+            set => SetValue(ShouldConvertNullToEmptyValueProperty, value);
+        }
+
+        public string RegexPattern
+        {
+            get => (string)GetValue(RegexPatternProperty);
+            set => SetValue(RegexPatternProperty, value);
+        }
+
+        public RegexOptions RegexOptions
+        {
+            get => (RegexOptions)GetValue(RegexOptionsProperty);
+            set => SetValue(RegexOptionsProperty, value);
+        }
+
+        protected virtual string DefaultRegexPattern => string.Empty;
+
+        protected virtual RegexOptions DefaultRegexOptions => RegexOptions.None;
+
+        protected override void OnAttachedTo(View bindable)
+        {
+            OnRegexPropertyChanged();
+            base.OnAttachedTo(bindable);
+        }
+
+        protected override object DecorateValue()
+        {
+            var value = base.DecorateValue()?.ToString();
+
+            if (ShouldTrimValue)
```

Comment:
Should we go for maximum flexibility and add the ability to only trim left/right or both? Can be in a follow up change

Reply:
I've come up with an idea:
think we can introduce additional enum

```
[Flags]
public enum TextValidationFlags {
    None = 0,
    TrimStart = 1,
    TrimEnd = 2,
    Trim = 3,
    NullToEmpty = 4
}
```

Thoughts?